Lib/test/test_typing.py

```diff
@@ -3809,6 +3809,26 @@ class Y(C[int]):
         self.assertEqual(Y.__qualname__,
                          'GenericTests.test_repr_2.<locals>.Y')
 
+    def test_repr_3(self):
+        T = TypeVar('T')
+        P = ParamSpec('P')
+
+        class MyCallable(Generic[P, T]):
+            pass
+
+        object_to_expected_repr = {
+            MyCallable[[], bool]:         "MyCallable[[], bool]",
+            MyCallable[[int], bool]:      "MyCallable[[int], bool]",
+            MyCallable[[int, str], bool]: "MyCallable[[int, str], bool]"
+        }
```

Comment:
I came up with some more diabolical cases. The behaviour looks correct to me in all of these (with your PR applied), but you could possibly add them as test cases:

```pycon
>>> from typing import *
>>> T, T1, T2 = TypeVar("T"), TypeVar("T1"), TypeVar("T2")
>>> P, P1, P2 = ParamSpec("P"), ParamSpec("P1"), ParamSpec("P2")
>>> Ts = TypeVarTuple("Ts")
>>> class MyCallable(Generic[P, T]): ...
...
>>> MyCallable[Concatenate[T1, P1], T2]
__main__.MyCallable[typing.Concatenate[~T1, ~P1], ~T2]
>>> _[int, Concatenate[*Ts, P2], str]
__main__.MyCallable[typing.Concatenate[int, *Ts, ~P2], str]
>>> _[bytes, list[int], list[str], [int, str, bytes]]
__main__.MyCallable[[int, bytes, list[int], list[str], int, str, bytes], str]
```

Reply:
Yes, I also came up with new cases. Updating!